Api reactions to message (#6730)
nextcloud/spreed: docs/reaction.md

# Reaction API

Base endpoint is: `/ocs/v2.php/apps/spreed/api/v1`

## React to a message

* Method: `POST`

> * Endpoint: `/chat/{token}/{messageId}`

Reviewer: This is not what your route file says

```suggestion
* Endpoint: `/reaction/{token}/{messageId}`
```

* Data:

    field | type | Description
    ---|---|---
    `reaction` | string | the reaction emoji

* Response:
    - Status code:
        + `201 Created`
        + `400 Bad Request` In case of any other error
        + `404 Not Found` When the conversation or message to react could not be found for the participant
        + `409 Conflict` User already did this reaction to this message

## Delete a reaction

* Method: `DELETE`

> * Endpoint: `/chat/{token}/{messageId}`

Reviewer: ```suggestion
* Endpoint: `/reaction/{token}/{messageId}`
```

* Data:

    field | type | Description
    ---|---|---
    `reaction` | string | the reaction emoji

* Response:
    - Status code:
        + `201 Created`
        + `400 Bad Request` In case of any other error
        + `404 Not Found` When the conversation or message to react or reaction could not be found for the participant

## Retrieve reactions of a message by type

* Method: `GET`

> * Endpoint: `/chat/{token}/{messageId}/?reaction=`

Reviewer: ```suggestion
* Endpoint: `/reaction/{token}/{messageId}`
```

# Reaction API

Base endpoint is: `/ocs/v2.php/apps/spreed/api/v1`

## React to a message

> * Method: `POST`

Reviewer: ```suggestion
* Required capability: `reactions`
* Method: `POST`
```
Same for all other methods in this doc

        + `400 Bad Request` In case of any other error
        + `404 Not Found` When the conversation or message to react could not be found for the participant
        + `409 Conflict` User already did this reaction to this message

## Delete a reaction

* Method: `DELETE`
* Endpoint: `/reaction/{token}/{messageId}`
* Data:

    field | type | Description
    ---|---|---
    `reaction` | string | the reaction emoji

* Response:
    - Status code:
        + `201 Created`
        + `400 Bad Request` In case of any other error
        + `404 Not Found` When the conversation or message to react or reaction could not be found for the participant

## Retrieve reactions of a message by type

* Method: `GET`
* Endpoint: `/reaction/{token}/{messageId}`
* Data:

    field | type | Description
    ---|---|---
    `reaction` | string | **Optional:** the reaction emoji

* Response:
    - Status code:
        + `200 OK`
        + `404 Not Found` When the conversation or message to react could not be found for the participant

    - Data:
        Array with data of reactions:

        field | type | Description
        ---|---|---
        `actorType` | string | `guests` or `users`

>         `actorId` | string | User id of the reaction author

Reviewer: ```suggestion
        `actorId` | string | Actor id of the reacting participant
```
